c3pzero_ignition/src/odom2tf_broadcaster_node.cpp

```diff
@@ -0,0 +1,58 @@
+/*******************************************************
+ * Copyright (C) 2022 Hyundai America Technical Center, Inc. (HATCI)
+ *
+ * This file can not be copied and/or distributed without the express
+ * permission of Hyundai America Technical Center, Inc. (HATCI)
+ *******************************************************/
+
+#include <memory>
+
+#include "geometry_msgs/msg/transform_stamped.hpp"
+#include "nav_msgs/msg/odometry.hpp"
+#include "tf2_ros/transform_broadcaster.h"
+#include <rclcpp/rclcpp.hpp>
+
+using std::placeholders::_1;
+
+/**
+ * @brief Simple class that broadcasts tf from the odometry message
+ *
+ */
+class TfBroadcaster : public rclcpp::Node
+{
+public:
+  TfBroadcaster() : Node("tf_broadcaster")
+  {
+    subscription_ = this->create_subscription<nav_msgs::msg::Odometry>(
+        "odom", 10, std::bind(&TfBroadcaster::topic_callback, this, _1));
+  }
+
+private:
+  void topic_callback(const nav_msgs::msg::Odometry::SharedPtr msg)
+  {
+    tf_broadcaster_ = std::make_shared<tf2_ros::TransformBroadcaster>(shared_from_this());
+    geometry_msgs::msg::TransformStamped transformStamped;
+
+    transformStamped.header = msg->header;
+    transformStamped.child_frame_id = msg->child_frame_id;
+    transformStamped.transform.translation.x = msg->pose.pose.position.x;
+    transformStamped.transform.translation.y = msg->pose.pose.position.y;
+    transformStamped.transform.translation.z = msg->pose.pose.position.z;
+    transformStamped.transform.rotation.x = msg->pose.pose.orientation.x;
+    transformStamped.transform.rotation.y = msg->pose.pose.orientation.y;
+    transformStamped.transform.rotation.z = msg->pose.pose.orientation.z;
+    transformStamped.transform.rotation.w = msg->pose.pose.orientation.w;
+
+    tf_broadcaster_->sendTransform(transformStamped);
+  }
+  rclcpp::Subscription<nav_msgs::msg::Odometry>::SharedPtr subscription_;
+  std::shared_ptr<tf2_ros::TransformBroadcaster> tf_broadcaster_;
+};
+
+int main(int argc, char* argv[])
+{
+  rclcpp::init(argc, argv);
+  rclcpp::spin(std::make_shared<TfBroadcaster>());
+  rclcpp::shutdown();
+  return 0;
```

Comment:
```suggestion
```
Just want to make sure everyone knows that `return 0` is implied if you finish running all the code in `main`. The standard guarantees that `0` gets returned so you never need to add that line yourself.